File: src/main/java/org/example/service/StoreService.java
package org.example.service;

import java.util.ArrayList;
import java.util.List;
import org.example.model.Product;

public class StoreService {
    private final List<Product> products = new ArrayList<>();

    public void initializeProducts() {
        // TODO: cargar productos iniciales
    }

    public List<Product> getProducts() {
        return new ArrayList<>(products);
    }

    public Product findProductById(int id) {
        for (Product product : products) {
            if (product.getId() == id) {
                return product;
            }
        }
        return null;
    }

    public boolean processPurchase(int productId, int quantity) {
        // TODO: validar stock y actualizar inventario
        return false;
    }
}

File: src/main/kotlin/org/example/service/StoreService.kt
package org.example.service

import org.example.model.Product

class StoreService {
    private val products = mutableListOf<Product>()

    fun initializeProducts() {
        // TODO: cargar productos iniciales
    }

    fun getProducts(): List<Product> = products.toList()

    fun findProductById(id: Int): Product? = products.find { it.id == id }

    fun processPurchase(productId: Int, quantity: Int): Boolean {
        // TODO: validar stock y actualizar inventario
        return false
    }
}

